lib/bookingsync/engine/account_model.rb

```diff
@@ -0,0 +1,114 @@
+class BookingsyncEngineUniquenessValidator < ActiveRecord::Validations::UniquenessValidator
+  def initialize(options)
+    if options[:scope].present? && options[:scope].respond_to?(:call)
+      options[:scope] = options[:scope].call
+    end
+    super
+  end
+end
+
+module BookingSync::Engine::AccountModel
+  extend ActiveSupport::Concern
+
+  included do
+    validates :synced_id, bookingsync_engine_uniqueness: {
+      scope: Proc.new { BookingSync::Engine.support_multi_applications? ? :host : nil }
+    }
+    scope :authorized, -> { where.not(oauth_access_token: nil) }
+  end
+
+  module ClassMethods
+    def from_omniauth(auth, host: nil)
```

Comment:
why not having two separate methods? It is really like a textbook violation of Interface Segregation Principle :P 

we should have `from_omniauth` and `from_omniauth_with_host` or something like that, we could have an extra method that would be checking which method should be called under the hood depending on the condition, but for sure that entire orchestration logic shouldn't be happening here.